### PR TITLE
Add C++ dev environment on CentOS using VirtualBox and Vagrant.

### PR DESCRIPTION
**Pull request by geirst**
@bjorncs and/or @vekterli please review
@aressem FYI

### CODE REVIEW

**Review by bjorncs — APPROVED** [review]
LGTM. Good work!

File: vagrant/README.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+### Connect to machine via SSH
+SSH agent forwarding is enabled to ensure easy interaction with GitHub inside the machine.
+
+    vagrant ssh
+
+### Checkout vespa source inside machine
+This is needed in order to compile and run tests fast on the local file system inside the machine.
+
+    git clone git@github.com:vespa-engine/vespa.git
```
Comment (vekterli) [inline vagrant/README.md]: If someone lands directly on this README-file, would be handy with a follow-up link on where to go from here to start building the thing.

**Review by vekterli — APPROVED** [review]
:+1:

**Comment by geirst** [comment]
@vekterli please review again

**Review by vekterli — APPROVED** [review]

**Pull request merged**